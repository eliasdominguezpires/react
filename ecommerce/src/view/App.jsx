
import React from 'react';
import { connect } from 'react-redux';

import { Container, Row, Col } from 'reactstrap';
import Main from '../AppRoutes';
import Header from './Header';

import 'bootstrap/dist/css/bootstrap.css';

const App = () => (
  <Container fluid>
    <Row className="justify-content-lg-center">
      <Col lg={10}>
        <Header />
      </Col>
    </Row>
    <Row className="justify-content-lg-center">
      <Col lg={10}>
        <Main />
      </Col>
    </Row>
    <Row>
      <p> Footer </p>
    </Row>
  </Container>
);

const mapStateToProps = state => ({
  ...state,
});

const mapDispatchToProps = () => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(App);
