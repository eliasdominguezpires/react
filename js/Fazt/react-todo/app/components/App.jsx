import React, { Component } from 'react';
import _ from 'lodash';

import TodoForm from './Todo/TodoForm';
import TodoList from './Todo/TodoList';

import { todos } from '../todos.json';

export default class App extends Component {

  constructor() {
    super();
    this.state = {
      todos
    };
  }

  render () {
    return (
      <div>
        <h1>React Todo</h1>
        <TodoForm
          todos={this.state.todos}
          createTask={this.createTask.bind(this)}
        />
        <TodoList
          todos={this.state.todos}
          toggleTask={this.toggleTask.bind(this)}
          saveTask={this.saveTask.bind(this)}
          deleteTask={this.deleteTask.bind(this)}
        />
      </div>
    )
  }

  createTask(task) {
    this.state.todos.push({
      task,
      isCompleted: false
    });
    this.setState({
      todos: this.state.todos
    })
  }

  toggleTask(task) {
    const foundTodo = _.find(
      this.state.todos,
      todo => todo.task === task
    );
    foundTodo.isCompleted = !foundTodo.isCompleted;
    this.setState({
      todos: this.state.todos
    });
  }

  saveTask(oldTask, newTask) {
    const foundTodo = _.find(
      this.state.todos,
      todo => todo.task === oldTask
    );
    foundTodo.task = newTask;
    this.setState({
      todos: this.state.todos
    })
  }

  deleteTask(task) {
    _.remove(this.state.todos, todo => todo.task === task);
    this.setState({
      todos: this.state.todos
    });
  }
}
