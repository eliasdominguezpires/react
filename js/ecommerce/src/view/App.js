import React, { Component } from 'react';
import { connect } from 'react-redux';
import Main from '../AppRoutes';
import Header from './Header';

import 'bootstrap/dist/css/bootstrap.css';

import { Container, Row, Col } from 'reactstrap';

class App extends Component {
  render() {
    return (
      <Container fluid>
        <Row className="justify-content-lg-center">
          <Col lg={10} >
            <Header />
          </Col>
        </Row>
        <Row className="justify-content-lg-center">
          <Col lg={10}>
            <Main />
          </Col>
        </Row>
        <Row>
        </Row>
      </Container>
    );
  };
}

const mapStateToProps = state => ({
  ...state
});

const mapDispatchToProps = dispatch => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(App);
