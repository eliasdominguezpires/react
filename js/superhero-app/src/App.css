.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.title {
  height: 22px;
  width: 121px;
  color: #000000;
  font-family: "SF Pro Text";
  font-size: 17px;
  font-weight: 600;
  letter-spacing: -0.41px;
  line-height: 22px;
  text-align: 'center';
}