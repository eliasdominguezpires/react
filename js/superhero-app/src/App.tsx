import React from 'react';
import './App.css';
import './style/Styles.css';
import { useSuperheroes } from './hooks/useSuperheroes';
import LoadingSpinner from './component/LoadingSpinner';
import { SuperheroInterface } from './interfaces/superheroInterface';

function App() {

  const { isLoading, superHeroes } = useSuperheroes();
  console.log(JSON.stringify(superHeroes, null, 5));

  const handleClick = (e: SuperheroInterface) => {
    console.log(JSON.stringify(e, null, 5));
    showNotification(e.name);
  }

  const showNotification = (name: string) => {
    Notification.requestPermission();
    new Notification(name);
  }

  return (
    <div className="Styles">
      <div className='navigation-bar-background'>
        <div className='title-header'>Superhero App</div>
      </div>
      <div className='row'>
        {isLoading ? <LoadingSpinner /> : superHeroes.map(item => (
          <div onClick={() => handleClick(item)}>
            <img src={item.images.sm} className="mask" alt={item.biography.fullName} />
            <div className='description-card'>
              <p> {item.name}</p>
              <p> {item.appearance.height[1]} - {item.appearance.weight[1]}</p>
            </div>
          </div>
        ))}

      </div>
    </div>
  );
}

export default App;
