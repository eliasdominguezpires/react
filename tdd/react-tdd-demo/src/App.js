import React, { Component } from 'react';
import './App.css';

const ENGLISH_TITLE = 'Hello, Paraguay!';
const SPANISH_TITLE = 'Â¡Hola, Paraguay!';
const PORTUGUES_TITLE = 'Ola, Brasil';

class App extends Component {
  state = {
    title: ENGLISH_TITLE,
  };

  handleTitleOnClick = () => {
    // toggle between the title language by updating the state    
    
    this.setState(
      state=>({        
        title: state.title === ENGLISH_TITLE?SPANISH_TITLE:ENGLISH_TITLE
      })
    )    
  }

  render() {
    return (
      <div className="App">
        <h1 id="title" onClick={this.handleTitleOnClick}>
          {this.state.title}
        </h1>
      </div>
    );
  }
}

export default App;
