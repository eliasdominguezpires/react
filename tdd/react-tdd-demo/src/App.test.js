import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('renders a header with the id "title"', () => {
    expect(wrapper.find('[id="title"]').length).toBe(1);
  });

  it('should toggle the language of the title on click', () => {
    const title = wrapper.find('[id="title"]');
    const englishTitle = 'Hello, Paraguay!'
    const spanishTitle = 'Â¡Hola, Paraguay!';

    expect(wrapper.state().title).toBe(englishTitle);
    title.simulate('click');
    expect(wrapper.state().title).toBe(spanishTitle);
  });
});
