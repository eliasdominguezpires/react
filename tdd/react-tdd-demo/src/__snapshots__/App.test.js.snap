// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a snapshot 1`] = `
<div
  className="App"
>
  <h1
    id="title"
    onClick={[Function]}
  >
    Hello, Paraguay!
  </h1>
</div>
`;
